src/overmap.cpp

```diff
@@ -434,6 +434,16 @@ const std::vector<overmap_special> &overmap_specials::get_all()
     return specials.get_all();
 }
 
+const overmap_special overmap_specials::get_specific( const std::string &id )
```

Comment:
```suggestion
const overmap_special overmap_specials::get_specific( const overmap_special_id &id )
```
And get rid of the call to `str()` in the loop below. `id` is an identifier referring to specific kind of thing: not a monster type, not a terrain type, but an overmap_special type.

----

The function isn't actually needed. You should use the established way to get the object that an id refers to: call `string_id::obj` or `string_id::operator*` - that should work for all types of ids:
```C++
const overmap_special_id id("foo");
const overmap_special &sp = *id;
```

Reply:
Thanks. That's much more cleaner.